app/main.py

```diff
@@ -1,8 +1,18 @@
 class Person:
-    # write your code here
-    pass
+    people = []
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people.append({"name": name, "age": age})
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    persons = []
+    for obj in people:
+        person = Person(obj["name"], obj["age"])
+        attribute = "wife" if "wife" in obj else "husband"
+        value = obj[attribute]
+        person.__setattr__(attribute, value)
```

Comment:
Using `__setattr__` to dynamically set attributes is valid, but it can be risky if the attribute names are not controlled or validated. Ensure that the input data is sanitized and that only expected attributes are set. Alternatively, consider using a more explicit method to handle optional attributes.